### PR TITLE
Major DiscordShardedClient update/maintaining

### PR DESCRIPTION
**Pull request by Plerx2493**
# Summary
Events were not updated in the sharded client

# Details
Some events were missing completely and some had missing implementation. Also the event renaming on the DiscordClient was not changed here. Created a "SGen" tool to reflect automaticly the events from the non-sharded client. @OoLunar wants to link this tool with a gh action in the future

# Note
I will look if i can make further updates

### CODE REVIEW

**Comment by OoLunar** [comment]
Please rebase master to fix the runner build errors

**Comment by OoLunar** [comment]
Also I might rewrite your `ShardedEventHandlingGen` tool later on to be easily reusable with more "sgen" scenarios

**Comment by Plerx2493** [comment]
> Also I might rewrite your `ShardedEventHandlingGen` tool later on to be easily reusable with more "sgen" scenarios

If you have specific changes in mind i would do it myself

**Comment by OoLunar** [comment]
Something like [this](https://github.com/OoLunar/HyperSharp/blob/master/src/HyperSharp.SourceGeneration/src/Program.cs), except maybe a little more fleshed out.

File: tools/DSharpPlus.Tools.ShardedEventHandlingGen/Program.cs [hunk 1]
```diff
@@ -0,0 +1,156 @@
+﻿using System.Globalization;
+using System.Reflection;
+using System.Text;
+using DSharpPlus;
+using Humanizer;
+
```
Comment (akiraveliara) [inline tools/DSharpPlus.Tools.ShardedEventHandlingGen/Program.cs:6]: is humanizer really needed here? also, extra newline

**Review by akiraveliara — APPROVED** [review]
unsure about humanizer, but looks good to me

File: tools/DSharpPlus.Tools.ShardedEventHandlingGen/Program.cs (hunk 1 above)
Comment (Plerx2493) [reply tools/DSharpPlus.Tools.ShardedEventHandlingGen/Program.cs:6]: I can remove it, added it for prototyping and i think `.Camelize` is the only method survived which is from it

**Comment by OoLunar** [comment]
This PR has a lot of potential in it. There are a few changes I'd like to make. I'd be okay with merging this as it is now though.

**Comment by akiraveliara** [comment]
fine to merge, @OoLunar ?

**Comment by OoLunar** [comment]
Yeah good to merge. It'll be a bit before I make my changes in a separate PR.

**Comment by Plerx2493** [comment]
wait

**Comment by Plerx2493** [comment]
private event names are now _EventName instead of _eventName

**Comment by Plerx2493** [comment]
I think i should fix this?

**Comment by OoLunar** [comment]
Please just sending everything in one comment, each of those comments are being sent to my email.

It's up to you if you want to fix it.

**Pull request merged**